File: CarHomeWidget/src/com/anod/car/home/skin/PropertiesFactory.java
package com.anod.car.home.skin;

import com.anod.car.home.prefs.PreferencesStorage;

public class PropertiesFactory {

	public static SkinProperties create(String skinName) {
		if (skinName.equals(PreferencesStorage.SKIN_WINDOWS7)) {
			return new MetroProperties();
		} else if (skinName.equals(PreferencesStorage.SKIN_HOLO)) {
			return new HoloProperties();
		} else if (skinName.equals(PreferencesStorage.SKIN_GLOSSY)) {
			return new GlossyProperties();
		} else if (skinName.equals(PreferencesStorage.SKIN_BBB)) {
			return new BBBProperties();
		}
		return new CarHomeProperties();
	}
}

File: CarHomeWidget/src/main/java/com/anod/car/home/skin/PropertiesFactory.kt
package com.anod.car.home.skin

import com.anod.car.home.prefs.preferences.Main

object PropertiesFactory {

    fun create(skinName: String): SkinProperties {
        return when (skinName) {
            Main.SKIN_WINDOWS7 -> MetroProperties()
            Main.SKIN_HOLO -> HoloProperties()
            Main.SKIN_GLOSSY -> GlossyProperties()
            Main.SKIN_BBB -> BBBProperties()
            Main.SKIN_CARDS -> CardsProperties()
            else -> CarHomeProperties()
        }
    }

}
